experiments/LES/surfacebubble.jl

```diff
@@ -0,0 +1,238 @@
+  "Prescribed MSEF Magnitude [W/m^2]"
+  F₀::FT
+  "Spatial Parameter [m]"
+  σ::FT
+  "Surface Heater Radius [m]"
+  a::FT
+  "Surface Heater Center [m]"
+  x₀::FT
+  "Time Cutoff [s]"
+  t₁::FT
+end
+function atmos_boundary_state!(nf::Union{NumericalFluxNonDiffusive,NumericalFluxGradient}, 
+                               bc::SurfaceDrivenBubbleBC, 
+                               m::AtmosModel,
+                               Y⁺::Vars, A⁺::Vars, 
+                               n⁻, 
+                               Y⁻::Vars, A⁻::Vars, 
+                               bctype, t,_...)
+  FT = eltype(Y⁺)
+  @inbounds begin
+    # Momentum b.c. prescribed (no flow across wall boundary)
+    if typeof(nf) <: NumericalFluxNonDiffusive
+      Y⁺.ρu -= 2 * dot(Y⁻.ρu, n⁻) * SVector(n⁻)
+    else
+      Y⁺.ρu -=  dot(Y⁻.ρu, n⁻) * SVector(n⁻)
+    end
+  end
+  nothing
+end
+function atmos_boundary_flux_diffusive!(nf::CentralNumericalFluxDiffusive, 
+                                        bc::SurfaceDrivenBubbleBC,
+                                        m::AtmosModel, 
+                                        F,
+                                        Y⁺::Vars, Σ⁺::Vars, A⁺::Vars, 
+                                        n⁻, 
+                                        Y⁻::Vars, Σ⁻::Vars, A⁻::Vars, 
+                                        bctype, t, _...)
+  FT = eltype(Y⁻)
+  @inbounds begin
+    # Momentum b.c. prescribed (no flow across wall boundary)
+    if typeof(nf) <: NumericalFluxNonDiffusive
+      Y⁺.ρu -= 2 * dot(Y⁻.ρu, n⁻) * SVector(n⁻)
+    else
+      Y⁺.ρu -=  dot(Y⁻.ρu, n⁻) * SVector(n⁻)
+    end
+  end
+
+  k̂  = vertical_unit_vector(m.orientation, A⁻)
+  r = sqrt((A⁻.coord[1]-bc.x₀)^2 + (A⁻.coord[2]-bc.x₀)^2) 
+  F₀ =  bc.F₀*(1 - sign(t-bc.t₁))/2 
+
+  @inbounds begin
+    # Energy flux prescribed (diffusive flux through bottom wall)
+    # MSEF ≡ Moist Static Energy Flux
+    if bctype == 1
+      if r > bc.a 
+        MSEF      = F₀ * exp(-(r-bc.a)^2 / bc.σ^2)
+        ∇h_tot⁺ = -Σ⁻.∇h_tot + 2 * MSEF * k̂
+      else
+        MSEF      = F₀
+        ∇h_tot⁺ = -Σ⁻.∇h_tot + 2 * MSEF * k̂
+      end
+    _,τ⁺ = turbulence_tensors(m.turbulence, Y⁺, Σ⁺, A⁺, t)
+    flux_diffusive!(m, F, Y⁺, τ⁺, ∇h_tot⁺)
+    end
+  end
+  nothing
+end
+
+boundary_state!(nf, m::AtmosModel, x...) =
+  atmos_boundary_state!(nf, m.boundarycondition, m, x...)
+boundary_flux_diffusive!(nf::NumericalFluxDiffusive,
+                         atmos::AtmosModel,
+                         F,
+                         Y⁺, Σ⁺, A⁺, n⁻,
+                         Y⁻, Σ⁻, A⁻,
+                         bctype, t,
+                         Y₁⁻, Σ₁⁻, A₁⁻) =
+  atmos_boundary_flux_diffusive!(nf, atmos.boundarycondition, atmos,
+                                 F,
+                                 Y⁺, Σ⁺, A⁺, n⁻,
+                                 Y⁻, Σ⁻, A⁻,
+                                 bctype, t,
+                                 Y₁⁻, Σ₁⁻, A₁⁻)
```

Comment:
Why is this needed in the driver? Seems to be duplicated code from the [AtmosModel](https://github.com/climate-machine/CLIMA/blob/c688b17762e3ef60233a46f6ba3fe094aa3bc2f6/src/Atmos/Model/AtmosModel.jl#L320-L339).

The same question applies to [dycoms](https://github.com/climate-machine/CLIMA/blob/c688b17762e3ef60233a46f6ba3fe094aa3bc2f6/experiments/LES/dycoms.jl#L142-L156)

Reply:
@jkozdon: fixed in both files.